oshmem/mca/sshmem/ucx/sshmem_ucx_module.c

```diff
@@ -120,15 +113,19 @@ segment_create_internal(map_segment_t *ds_buf, void *address, size_t size,
 
     mem_map_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
                                 UCP_MEM_MAP_PARAM_FIELD_LENGTH |
-                                UCP_MEM_MAP_PARAM_FIELD_FLAGS;
+                                UCP_MEM_MAP_PARAM_FIELD_FLAGS  |
+                                UCP_MEM_MAP_PARAM_FIELD_MEMORY_TYPE;
 
-    mem_map_params.address    = address;
-    mem_map_params.length     = size;
-    mem_map_params.flags      = flags;
+    mem_map_params.address     = address;
+    mem_map_params.length      = size;
+    mem_map_params.flags       = flags;
+    mem_map_params.memory_type = mem_type;
 
     status = ucp_mem_map(spml->ucp_context, &mem_map_params, &mem_h);
     if (UCS_OK != status) {
-        SSHMEM_ERROR("ucp_mem_map() failed: %s\n", ucs_status_string(status));
+        SSHMEM_ERROR("Failed to allocate DEVICE_NIC_MEM"
```

Comment:
I'm not sure if this error msg is indicative.
In this particular context, (in theory) it might not necessarily be DEVICE_NIC_MEM...

Reply:
```suggestion
        SSHMEM_ERROR("ucp_mem_map() failed: %s %s\n",
                     ucs_status_string(status),
                     mem_type == UCS_MEMORY_TYPE_RDMA ?
                     "failed to allocate DEVICE_NIC_MEM, falls back to host memory" : "");
```
Maybe that way?